File: backend/zhijian-order/src/main/java/com/zhijian/pojo/Order.java
package com.zhijian.pojo;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 订单实体类。
 */
@Data
@TableName("oms_order")
public class Order implements Serializable {

    /**
     * 订单 ID。
     */
    @TableId
    private Long id;

    /**
     * 用户 ID。
     */
    private Long userId;

    /**
     * 商家 ID。
     */
    private Long sellerId;

    /**
     * 订单号。
     */
    private String orderNo;

    /**
     * 药品 ID。
     */
    @TableField(exist = false)
    private Long medicineId;

    /**
     * 药品名称。
     */
    @TableField(exist = false)
    private String medicineName;

    /**
     * 药品图片。
     */
    @TableField(exist = false)
    private String medicineImage;

    /**
     * 订单项列表。
     */
    @TableField(exist = false)
    private List<OrderItem> items;

    /**
     * 购买数量。
     */
    @TableField(exist = false)
    private Integer quantity;

    /**
     * 单价。
     */
    @TableField(exist = false)
    private BigDecimal price;

    /**
     * 商品总金额。
     */
    private BigDecimal totalAmount;

    /**
     * 优惠券抵扣金额。
     */
    private BigDecimal couponAmount;

    /**
     * 实付金额。
     */
    private BigDecimal payAmount;

    /**
     * 优惠券记录 ID。
     */
    private Long couponHistoryId;

    /**
     * 订单状态。
     */
    private Integer status;

    /**
     * 收货人姓名。
     */
    private String receiverName;

    /**
     * 收货人电话。
     */
    private String receiverPhone;

    /**
     * 收货地址。
     */
    private String receiverAddress;

    /**
     * 退款原因。
     */
    private String refundReason;

    /**
     * 退款备注。
     */
    private String refundRemark;

    /**
     * 审核不通过原因。
     */
    private String auditReason;

    /**
     * 处方图片。
     */
    private String prescriptionImage;

    /**
     * 创建时间。
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间。
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 支付时间。
     */
    @TableField("payment_time")
    private LocalDateTime payTime;

    /**
     * 发货时间。
     */
    private LocalDateTime deliveryTime;

    /**
     * 完成时间。
     */
    private LocalDateTime finishTime;

    /**
     * 评价状态。
     */
    private Integer commentStatus;

    /**
     * 就诊人 ID。
     */
    private Long patientId;

    /**
     * 药师审核状态。
     */
    @TableField("pharmacist_audit_status")
    private Integer auditStatus;
}

File: frontend/apps/admin/src/services/order.ts
import request from './request';

// 管理端订单管理接口。
export interface OrderQueryDTO {
  page: number;
  size: number;
  status?: number;
}

export interface Order {
  id: number;
  orderNo: string;
  medicineName?: string;
  medicineImage?: string;
  price?: number;
  quantity?: number;
  totalAmount: number;
  payAmount: number;
  status: number; // 0:待支付 1:待发货 2:已发货 3:已完成 4:售后中 5:已退款 -1:已取消 7:待审核
  createTime: string;
  receiverName?: string;
  receiverPhone?: string;
  receiverAddress?: string;
  sellerId?: number;
  userId?: number;
  auditStatus?: number;
  auditReason?: string;
  payTime?: string;
  deliveryTime?: string;
  finishTime?: string;
  couponAmount?: number;
  prescriptionImage?: string;
}

export interface PageResult<T> {
  records: T[];
  total: number;
  size: number;
  current: number;
  pages: number;
}

// 获取管理端订单列表。
export const getAdminOrderList = (params: OrderQueryDTO) => {
  return request.get<PageResult<Order>>('/orders/admin/list', { params });
};

export interface OrderAuditDTO {
  orderId: number;
  pass: boolean;
  reason?: string;
}

// 审核处方或待审核订单。
export const auditOrder = (data: OrderAuditDTO) => {
  return request.post('/orders/audit', data);
};
